### PR TITLE
[JENKINS-32493] Baseline 1.580.1 and Parent POM 2.3

### PR DESCRIPTION
**Pull request by andresrc**
* [x] Update baseline to 1.580.1
* [x] Update parent POM to 2.3
* [x] Adapt to JTH 2.1 and its updated HtmlUnit

@reviewbybees

### CODE REVIEW

**Comment by kwhetstone** [comment]
This change really simplifies the pom.  However, I notice that findbugs-maven-plugin.version is no longer configurable.  Is this something that's going to be missed with the new configuration?  I'm not sure if people are using this so it might be a non-issue.

**Comment by andresrc** [comment]
@kwhetstone I don't have a strong opinion on that and would like to hear more. But my current position with regard to findbugs, and in the spirit of minimizing switches, is that we should aim for plugins using the "recommended" one. On the other side, I've seen many plugins configuring the possibility of not failing the build with findbugs errors, to handle transient cases. I think and including that switch in the parent pom, though not encouraging its use :smile:

**Comment by stephenc** [comment]
@kwhetstone a property like `findbugs-maven-plugin.version` which is only used in one and only one place is an anti-pattern... it would be different if, for example, the maven plugin had the same release lifecycle as the annotations that it checks against and we therefore had to keep the annotations dependency in lock step with the maven plugin... in general:

* where there are two or more things that have to be kept in lock step, then you need to use a property
* where other functionality drives needing to configure from the CLI, then you need to use a property... the example is letting developers easily test their plugin against different versions of jenkins... `mvn -Djenkins.version=... hpi:run` in that case you may occasionally find that a newer version of jenkins has a requirement for a newer version of the hpi plugin... and more rarely a newer version of the stapler plugin... so in order to support the use case of letting you use `hpi:run` against different versions of jenkins we need to let the hpi plugin and the stapler plugin versions be controlled from the CLI.
* In all other cases, if the version is only defined in one place, *do not use a property*. Plugins that need to override the version can do so *the maven way* by overriding the version explicitly in `pluginManagement`

**Comment by kwhetstone** [comment]
@stephenc thanks for the explanation.  I was primarily concerned about the case where you're using the cli since someone might have been using that variable for something.  I'm all for doing things the maven way.

File: pom.xml [hunk 1]
```diff
@@ -12,7 +13,10 @@
     <description>Allows Jenkins administrators to control what in-process scripts can be run by less-privileged users.</description>
     <url>https://wiki.jenkins-ci.org/display/JENKINS/Script+Security+Plugin</url> 
     <properties>
-      <findbugs-maven-plugin.version>3.0.2</findbugs-maven-plugin.version>
+      <jenkins.version>1.509.4</jenkins.version>
+      <java.level>6</java.level>
+      <!-- TODO check if we'd better add the PI. -->
+      <jelly.requirePI>false</jelly.requirePI>
```
Comment (jglick) [inline pom.xml]: Yes you should.

File: pom.xml [hunk 2]
```diff
@@ -28,19 +32,34 @@
...
+      <id>repo.jenkins-ci.org</id>
+      <url>http://repo.jenkins-ci.org/public/</url>
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </pluginRepository>
+  </pluginRepositories>
```
Comment (jglick) [inline pom.xml]: What was wrong with the original versions of these sections?

File: pom.xml [hunk 3]
```diff
@@ -55,8 +74,6 @@
         <dependency>
             <groupId>com.google.code.findbugs</groupId>
             <artifactId>annotations</artifactId>
-            <version>3.0.0</version>
-            <scope>provided</scope>
         </dependency>
         <dependency> <!-- TODO default as of 1.568+ -->
```
Comment (jglick) [inline pom.xml]: Is this not now included by default in the parent POM? IMO it should be.

File: pom.xml [hunk 4]
```diff
@@ -12,7 +13,10 @@
     <description>Allows Jenkins administrators to control what in-process scripts can be run by less-privileged users.</description>
     <url>https://wiki.jenkins-ci.org/display/JENKINS/Script+Security+Plugin</url> 
     <properties>
-      <findbugs-maven-plugin.version>3.0.2</findbugs-maven-plugin.version>
+      <jenkins.version>1.509.4</jenkins.version>
```
Comment (jglick) [inline pom.xml]: This is too old. I just checked and found that the oldest baseline used by a dependency of this plugin is now 1.580.1, so I might as well update to that here too. Will do so in a separate PR.

Comment (jglick) [inline pom.xml]: …#38

**Comment by andresrc** [comment]
Addressed feedback and rebased.

File: pom.xml [hunk 5]
```diff
@@ -12,7 +13,10 @@
     <description>Allows Jenkins administrators to control what in-process scripts can be run by less-privileged users.</description>
     <url>https://wiki.jenkins-ci.org/display/JENKINS/Script+Security+Plugin</url> 
     <properties>
-      <findbugs-maven-plugin.version>3.0.2</findbugs-maven-plugin.version>
+      <jenkins.version>1.580.1</jenkins.version>
+      <!-- TODO: Compilation errors to investigate when using 2.0 (matrix related) -->
+      <jenkins-test-harness.version>1.580.1</jenkins-test-harness.version>
```
Comment (jglick) [inline pom.xml]: :bug: probably you just need to add a test-scoped dependency on `matrix-auth`.

File: src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApprovalTest.java [hunk 6]
```diff
@@ -179,4 +178,19 @@ public boolean handleConfirm(Page page, String message) {
...
     }
 
+    private void clickAndWait(HtmlElement e) throws IOException {
+        e.click();
+        e.getPage().getWebClient().waitForBackgroundJavaScript(10000);
+    }
+
+    private void click(HtmlPage page, String id, String value) throws IOException {
+        for (HtmlElement e : page.getElementById(id).getElementsByTagName("button")) {
+            if (e.hasAttribute("class") && value.equals(e.getAttribute("class"))) {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApprovalTest.java]: Yuck. Would it be easier to test if the script approval page just added some more `id`s?

Comment (andresrc) [inline src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApprovalTest.java]: iirc it also tests it is removed from the pending list...

In any case the changes only tried to make it compatible with the upgraded HTMLUnit, not to refactor the test.

Thanks!

**Comment by jglick** [comment]
> Update baseline to 1.580.1

is misleading, since that was already the baseline.

**Pull request merged**